streams/src/main/java/org/apache/kafka/streams/processor/internals/ActiveTaskCreator.java

```diff
@@ -136,8 +135,8 @@ StreamsProducer streamsProducerForTask(final TaskId taskId) {
     }
 
     StreamsProducer threadProducer() {
-        if (processingMode != EXACTLY_ONCE_V2) {
-            throw new IllegalStateException("Expected EXACTLY_ONCE_V2 to be enabled, but the processing mode was " + processingMode);
+        if (processingMode == EXACTLY_ONCE_ALPHA) {
```

Comment:
I'm not sure I understand why this didn't always include ALOS.

To make sure this could always return the producer for ALOS we just never needed it to before and nothing changed in this PR to let it work with ALOS?